engine/server/engine/main.go

```diff
@@ -338,3 +359,87 @@ func formatFilenameFunctionForLogs(filename string, functionName string) string
 	output.WriteString("]")
 	return output.String()
 }
+
+func restApiServer(
+	ctx context.Context,
+	serverArgs *args.EngineServerArgs,
+	enclave_manager *enclave_manager.EnclaveManager,
+	perWeekLogsDatabaseClient centralized_logs.LogsDatabaseClient,
+	perFileLogsDatabaseClient centralized_logs.LogsDatabaseClient,
+	logFileManager *log_file_manager.LogFileManager,
+	metricsClient metrics_client.MetricsClient,
+) {
+	logrus.Info("Running REST API server...")
+
+	// This is how you set up a basic Echo router
+	e := echo.New()
+	// Log all requests
+	e.Use(echomiddleware.Logger())
+
+	// ============================== Engine Management API ======================================
+	// OpenAPI schema.
+	swagger_engine, err := engineApi.GetSwagger()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error loading swagger spec\n: %s", err)
+		os.Exit(1)
```

Comment:
what will happen in this case? will the other engine servers gracefully shut down?